alpenhorn/client.py

```diff
@@ -666,26 +668,40 @@ def format_transport(serial_num):
 
     # Figure out if it is formatted.
     print("Checking to see if disc is formatted. Please wait.")
-    fp = os.popen("parted -s %s print" % dev)
-    formatted = False
-    part_start = False
-    while True:
-        l = fp.readline()
-        if not l:
-            break
-        if l.find("Number") == 0 and l.find("Start") > 0 and l.find("File system") > 0:
-            part_start = True
-        elif l.strip() != "" and part_start:
+    try:
+        formatted = False
```

Comment:
Since we can assume this assignment will never raise `subprocess.CalledProcessError`, it should be done outside the try block.